src/rhsm/connection.py

```diff
@@ -1361,7 +1361,7 @@ def regenEntitlementCertificates(self, consumer_id, lazy_regen=True):
             result = True
         except (RemoteServerException, httpslib.BadStatusLine) as e:
             # 404s indicate that the service is unsupported (Candlepin too old, or SAM)
-            if isinstance(e, httpslib.BadStatusLine) or e.code == "404":
+            if isinstance(e, httpslib.BadStatusLine) or str(e.code) == "404":
```

Comment:
If the 'raise e' in the non-404 cases just does 'raise, usually the stack trace will be more useful (it will include the original stack, not the stack from where 'raise e' happened):

``` python
            else:
                 # Something else happened that we should probabaly raise
                 raise
```